linera-core/src/updater.rs

```diff
@@ -167,13 +170,15 @@ where
                 }
             }
         }
-        // If a key reaches a quorum or it becomes clear that no key can, wait for the grace
-        // period to collect more values or error information and then stop.
-        if end_time.is_none()
-            && (highest_key_score >= committee.quorum_threshold()
-                || highest_key_score + remaining_votes < committee.quorum_threshold())
-        {
-            end_time = Some(Instant::now() + start_time.elapsed().mul_f64(GRACE_PERIOD));
+        // If it becomes clear that no key can reach a quorum, break early.
+        if highest_key_score + remaining_votes < committee.quorum_threshold() {
+            break 'vote_wait;
```

Comment:
cc @afck for reviewing this part

Reply:
Makes sense to me. :+1: 
(An alternative would be to set `end_time` to `now()`, but maybe the `break` is cleaner.)